### PR TITLE
create estimate fee example

### LINKED ISSUE #505: how to use EstimateFee
Can someone give an example to use it? 
I don’t know how to convert it into the required format. I think usage examples should not directly use maxfee and omit the step of calculating fee.I need an example.

### PR DESCRIPTION
**Pull request by rianhughes**
closes https://github.com/NethermindEth/starknet.go/issues/505

### CODE REVIEW

**Comment by KillerJi** [comment]
Thanks sir.
I used my own goerli node of infura starknet and followed the steps in the readme, but it seems that there is a problem with the parameters. Have you tried it and there is no problem? Error reported:
root@J:~/code/starknet.go/examples/estimateFee# go run main.go
Starting estimateFee example
Established connection with the client
Building transaction
Calling EstimateFee
panic: Failed to estimate fee: Invalid params

goroutine 1 [running]:
main.main()
         /root/code/starknet.go/examples/estimateFee/main.go:102 +0x7e5
exit status 2

**Comment by rianhughes** [comment]
> Thanks sir. I used my own goerli node of infura starknet and followed the steps in the readme, but it seems that there is ..

Yes, the example is still working for me when running against a node running RPCv06. It may be that infura are running an old version of the RPC.

Try running against the free goerli endpoint here https://data.voyager.online/

**Comment by KillerJi** [comment]
> > Thanks sir. I used my own goerli node of infura starknet and followed the steps in the readme, but it seems that there is ..
> 
> Yes, the example is still working for me when running against a node running RPCv06. It may be that infura are running an old version of the RPC.
> 
> Try running against the free goerli endpoint here https://data.voyager.online/

I tried your method and done finally! thank you so much sir

File: examples/estimateFee/go.mod [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module account
+
+go 1.21
+
+require (
+	github.com/NethermindEth/starknet.go v0.4.6-0.20231005024141-742a82479868
```
Comment (IronGauntlets) [inline examples/estimateFee/go.mod:6]: Is this version of starknet.go automatically updated?

File: examples/estimateFee/go.work [hunk 2]
```diff
@@ -0,0 +1,6 @@
+go 1.21
+
+use (
+	.
+	../..
+)
```
Comment (IronGauntlets) [inline examples/estimateFee/go.work:6]: In order to create this file did you run the following command: `go work init . ../..`?

File: examples/estimateFee/go.mod (hunk 1 above)
Comment (rianhughes) [reply examples/estimateFee/go.mod:6]: Nope.

File: examples/estimateFee/go.work (hunk 2 above)
Comment (rianhughes) [reply examples/estimateFee/go.work:6]: I believe this file came with one of the initial examples before we took over the project.

**Review by IronGauntlets — APPROVED** [review]

**Comment by KillerJi** [comment]
Hi sir 

I want to estimate the transfer cost. I added 2 parameters but got an error.

	acc, err := new(felt.Felt).SetString(account_addr)
	if err != nil {
		panic(err.Error())
	}
	transferNum, err := new(felt.Felt).SetString("100000000000")
	if err != nil {
		panic(err.Error())
	}
	FnCall := rpc.FunctionCall{
		ContractAddress:    contractAddress,                               //contractAddress is the contract that we want to call
		EntryPointSelector: utils.GetSelectorFromNameFelt(contractMethod), //this is the function that we want to call
		Calldata:           []*felt.Felt{acc, transferNum},
	}
	// Build the Calldata
	InvokeTx.Calldata, err = accnt.FmtCalldata([]rpc.FunctionCall{FnCall})
	if err != nil {
		panic(err.Error())
	}

	// Sign the transaction
	err = accnt.SignInvokeTransaction(context.Background(), &InvokeTx)
	if err != nil {
		panic(err.Error())
	}

	fmt.Println("Calling EstimateFee")
	resp, err := accnt.EstimateFee(context.Background(), []rpc.BroadcastTxn{InvokeTx}, []rpc.SimulationFlag{}, rpc.BlockID{Tag: "latest"})
	if err != nil {
		panic(fmt.Sprintf("Failed to estimate fee: %v", err))
	}

I have tried many times but it still doesn’t work. This address is the test address and the balance is sufficient. What’s wrong?
error info:

Starting estimateFee example
Established connection with the client
Building transaction
Calling EstimateFee
panic: Failed to estimate fee: Transaction execution error

**Pull request merged**